File: apps/web/tests/unit/trading-mode.test.ts
import { describe, expect, it } from "vitest";
import { resolveTradingMode, isLiveTradingEnabled } from "@/lib/trading-mode";

describe("resolveTradingMode (fail-closed)", () => {
  it("maps 'disabled' to PAPER, never LIVE", () => {
    expect(resolveTradingMode("disabled")).toBe("PAPER");
  });

  it("maps 'enabled' to LIVE", () => {
    expect(resolveTradingMode("enabled")).toBe("LIVE");
  });

  it("maps unknown/missing values to UNKNOWN, never LIVE", () => {
    expect(resolveTradingMode("bogus")).toBe("UNKNOWN");
    expect(resolveTradingMode(null)).toBe("UNKNOWN");
    expect(resolveTradingMode(undefined)).toBe("UNKNOWN");
  });

  it("never enables live for anything but the exact 'enabled' string", () => {
    expect(isLiveTradingEnabled("disabled")).toBe(false);
    expect(isLiveTradingEnabled("DISABLED")).toBe(false);
    expect(isLiveTradingEnabled("1")).toBe(false);
    expect(isLiveTradingEnabled("enabled")).toBe(true);
  });
});

File: apps/desktop/test/core/trading_mode_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:alpha_algo_desktop/core/trading_mode.dart';

void main() {
  group('resolveTradingMode (fail-closed)', () {
    test("maps 'disabled' to paper, never live", () {
      expect(resolveTradingMode('disabled'), TradingMode.paper);
    });

    test("maps 'enabled' to live", () {
      expect(resolveTradingMode('enabled'), TradingMode.live);
    });

    test('maps unknown/missing to unknown, never live', () {
      expect(resolveTradingMode('bogus'), TradingMode.unknown);
      expect(resolveTradingMode(null), TradingMode.unknown);
    });

    test('never enables live for anything but the exact string', () {
      expect(isLiveTradingEnabled('disabled'), isFalse);
      expect(isLiveTradingEnabled('DISABLED'), isFalse);
      expect(isLiveTradingEnabled('1'), isFalse);
      expect(isLiveTradingEnabled('enabled'), isTrue);
    });
  });
}
